DigitalLearningSolutions.Data/Services/LearningHubResourceService.cs

```diff
@@ -0,0 +1,103 @@
+            this.learningHubApiClient = learningHubApiClient;
+            this.learningResourceReferenceDataService = learningResourceReferenceDataService;
+            this.logger = logger;
+        }
+
+        public async Task<LearningResourceReferenceWithResourceDetails?> GetResourceByReferenceId(
+            int resourceReferenceId
+        )
+        {
+            try
+            {
+                // TODO HEEDLS-747: tackle different response codes rather than catching all exceptions
+                var upToDateResourceDetails = await learningHubApiClient.GetResourceByReferenceId(resourceReferenceId);
+                return new LearningResourceReferenceWithResourceDetails(upToDateResourceDetails, false);
+            }
+            catch (Exception e)
+            {
+                logger.LogWarning(
+                    $"Call to Learning Hub Open API failed when trying to call /Resource/{resourceReferenceId} endpoint. " +
+                    $"Using fall-back data instead. Exception: {e.Message}"
+                );
+
+                var fallBackResourceDetails =
+                    learningResourceReferenceDataService.GetResourceReferenceDetailsByReferenceIds(
+                        new[] { resourceReferenceId }
+                    ).SingleOrDefault();
+
+                return fallBackResourceDetails == null
+                    ? null
+                    : new LearningResourceReferenceWithResourceDetails(
+                        fallBackResourceDetails,
+                        true
+                    );
+            }
+        }
+
+        public async Task<BulkLearningResourceReferences> GetBulkResourcesByReferenceIds(
+            IList<int> resourceReferenceIds
+        )
+        {
+            try
+            {
+                // TODO HEEDLS-747: tackle different response codes rather than catching all exceptions
+                var upToDateResourceDetails =
+                    await learningHubApiClient.GetBulkResourcesByReferenceIds(resourceReferenceIds);
+                return new BulkLearningResourceReferences(upToDateResourceDetails, false);
+            }
+            catch (Exception e)
+            {
+                logger.LogWarning(
+                    "Call to Learning Hub Open API failed when trying to call /Resource/Bulk endpoint. " +
+                    $"Using fall-back data instead. Exception: {e.Message}"
+                );
+
+                var fallBackResources =
+                    learningResourceReferenceDataService.GetResourceReferenceDetailsByReferenceIds(resourceReferenceIds)
+                        .ToList();
+
+                var bulkResourceReferences = new BulkResourceReferences
+                {
+                    ResourceReferences = fallBackResources,
+                    UnmatchedResourceReferenceIds = resourceReferenceIds.Where(
+                        refId => fallBackResources.All(resource => resource.RefId != refId)
+                    ).ToList(),

```

Comment:
Maybe slightly clearer to use [Except()](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.except?view=net-6.0), but there's not much in it.